### PR TITLE
New patch

### PR DESCRIPTION
**Pull request by raffaelespazzoli**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,9 +13,23 @@ This library layers on top of the Operator SDK and with the objective of helping
...
-2. [Basic operator lifecycle](#Basic-Operator-Lifecycle-Management) needs (validation, initialization, status and error management, finalization)
-3. [Enforcing resources operator support](#Enforcing-Resource-Operator-Support). For those operators which calculate a set of resources that need to exist and then enforce them, generalized support for the enforcing phase is provided.
+1. [Utility Method](#Utility-Methods) Utility methods that are callable by any operator.
+2. [Idempotent methods](#Idempotent-Methods-to-Manipulate-Resources) to manipulate resources and arrays of resources
+3. [Basic operator lifecycle](#Basic-Operator-Lifecycle-Management) needs (validation, initialization, status and error management, finalization)
+4. [Enforcing resources operator support](#Enforcing-Resource-Operator-Support). For those operators which calculate a set of resources that need to exist and then enforce them, generalized support for the enforcing phase is provided.
+
+## Utility Methods
+
+Prior to version v2.x the general philosophy of this library was that new operator would inherint from REcorncileBase and in doing so they would havbe access to a bunch of utility methods.
```
Comment (sabre1041) [inline README.md:23]:
```suggestion
Prior to version v2.x the general philosophy of this library was that new operator would inherent from ReconcileBase and in doing so they would have access to a bunch of utility methods.
```

File: README.md [hunk 2]
```diff
@@ -13,9 +13,23 @@ This library layers on top of the Operator SDK and with the objective of helping
...
+With release v2.0.0 a new approach is avaibale. Utility methods are callable by any operator havintg to inehrit. This makes it easier to use this library and does not conflict with autogenerate code from kube-builder and operator-sdk.
```
Comment (sabre1041) [inline README.md:24]:
```suggestion
With release v2.0.0, a new approach is available. Utility methods are callable by any operator having to inherit. This makes it easier to use this library and does not conflict with autogenerate code from `kube-builder` and `operator-sdk`.
```

File: README.md [hunk 3]
```diff
@@ -13,9 +13,23 @@ This library layers on top of the Operator SDK and with the objective of helping
...
+Most of the Utility methods receive a context.Context parameter. Normally this context must be initialized with a `logr.Logger` and a `rest.Config`. Some utility methods may require more, see each individual documentation.
```
Comment (sabre1041) [inline README.md:25]:
```suggestion
Most of the Utility methods receive a `context.Context` parameter. Normally this context must be initialized with a `logr.Logger` and a `rest.Config`. Some utility methods may require more, see each individual documentation.
```

File: README.md [hunk 4]
```diff
@@ -13,9 +13,23 @@ This library layers on top of the Operator SDK and with the objective of helping
...
+
+Utility methods are current organized in the following folders:
```
Comment (sabre1041) [inline README.md:27]:
```suggestion
Utility methods are currently organized in the following folders:
```

File: README.md [hunk 5]
```diff
@@ -13,9 +13,23 @@ This library layers on top of the Operator SDK and with the objective of helping
...
+
+1. crud: idempotent create/update/delete functions.
+2. discoveryclient: methods related to the discovery client, topically used to load `apiResource` objects.
```
Comment (sabre1041) [inline README.md:30]:
```suggestion
2. discoveryclient: methods related to the discovery client, typically used to load `apiResource` objects.
```

File: pkg/util/reconciler.go [hunk 6]
```diff
@@ -112,78 +111,79 @@ func (r *ReconcilerBase) GetDiscoveryClient() (*discovery.DiscoveryClient, error
...
 
-// GetDynamicClientOnAPIResource returns a dynamic client on an APIResource. This client can be further namespaced.
-func (r *ReconcilerBase) GetDynamicClientOnAPIResource(resource metav1.APIResource) (dynamic.NamespaceableResourceInterface, error) {
-	return r.getDynamicClientOnGVR(schema.GroupVersionResource{
-		Group:    resource.Group,
-		Version:  resource.Version,
-		Resource: resource.Name,
-	})
-}
+// // GetDynamicClientOnAPIResource returns a dynamic client on an APIResource. This client can be further namespaced.
```
Comment (sabre1041) [inline pkg/util/reconciler.go:114]: Do we want to remove all of this commented out code?

**Review by sabre1041 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (raffaelespazzoli) [reply README.md:23]: can you fix from inherent to inherit?

File: README.md (hunk 2 above)
Comment (raffaelespazzoli) [reply README.md:24]: can you change from `having` to `without having`?

File: README.md (hunk 1 above)
Comment (raffaelespazzoli) [reply README.md:23]: fixed

File: README.md (hunk 2 above)
Comment (raffaelespazzoli) [reply README.md:24]: fixed

File: pkg/util/reconciler.go (hunk 6 above)
Comment (raffaelespazzoli) [reply pkg/util/reconciler.go:114]: fixed

**Comment by raffaelespazzoli** [comment]
all comments should have been addressed @sabre1041

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**